pyproject.toml

```diff
@@ -1,2 +1,85 @@
 [tool.black]
 line-length = 120
+
+[tool.pylint."MESSAGES CONTROL"]
+disable = [
+    "arguments-differ",
+    "broad-except",
+    # "broad-exception-raised",          # Only available in later pylint
+    "c-extension-no-member",             # seems to be weird false pos
+    "consider-iterating-dictionary",     # lol no
+    "consider-using-f-string",           # don't care
+    "duplicate-code",                    # dont' care
+    "eval-used",                         # needed
+    "exec-used",                         # needed
+    "fixme",
+    "global-statement",                  # needed
+    "import-outside-toplevel",           # needed
+    "inconsistent-return-statements",    # don't care
+    "invalid-name",                      # SUPER don't care
+    "keyword-arg-before-vararg",         # literally nonsensical
+    "line-too-long",                     # black takes care of my line lengths
+    "missing-class-docstring",
+    "missing-function-docstring",
+    "missing-module-docstring",
+    "missing-timeout",                   # TODO: fix
+    "no-else-break",                     # i prefer this actually
+    "no-else-continue",                  # ditto
+    "no-else-raise",                     # ditto
+    "no-else-return",                    # ditto
+    "pointless-string-statement",        # fine as alt comment syntax
+    "redefined-builtin",
+    "superfluous-parens",                # don't care
+    "too-few-public-methods",            # dumb
+    "too-many-arguments",                # dumb
+    "too-many-boolean-expressions",      # needed
+    "too-many-branches",                 # needed
+    "too-many-instance-attributes",      # dumb
+    "too-many-lines",
+    "too-many-locals",                   # dumb
+    "too-many-nested-blocks",
+    "too-many-return-statements",
+    "too-many-statements",
+    "unsubscriptable-object",            # false positives
+    "unsupported-binary-operation",      # false pos on type sigs, plus mypy catches anyway
+    "use-dict-literal",                  # don't care
+    "useless-return",                    # need it for mypy
+]
+
+[tool.ruff]
+target-version = "py37"
+ignore = [
+    "E501",
```

Comment:
```suggestion
```
`E501` and `line-length` both affect the same test.  It is best to set `line-length=447` (below) so that contributors who add even longer lines need to justify their decision in the PR review process.

Reply:
Nah, I still think max line lengths are a fake idea. `black` enforces reasonable line-breaking practices by itself; the remaining long lines are due to long strings, and I don't think that manually breaking those into separate strings is worth anyone's time or effort. (In fact I think it's *actively a bad idea*, as changes to the string might require cascading edits to rejigger the line lengths. I follow [Semantic Line Breaks](https://sembr.org/) in most of my projects, and where that doesn't make sense, rely on soft-wrapping.